space-plugins/nuxt-base/server/utils/space.ts

```diff
@@ -1,10 +1,18 @@
+import { getManagementBaseUrl, getRegion } from '@storyblok/region-helper';
 import { FetchSpaceInfo, SpaceInfo } from '~/types/space';
 
 export const fetchSpaceInfo: FetchSpaceInfo = async ({
 	spaceId,
 	accessToken,
 }) => {
-	const apiHost = getManagementApiHost(spaceId);
+	const region = getRegion(spaceId);
```

Comment:
The previous implementation of `getManagementApiHost` was falling back to `eu` region, but I don't think that's a correct implementation. Instead, I'm returning a specific error code here. What do you think?

Reply:
Looks nice to me 💯 

What if these utils methods that use the MAPI return something like:

```js
export const fetchSpaceInfo: FetchSpaceInfo | ManagementApiError = async ({
```

and we create this `ManagementApiError` type so it can be shared across with the other utility methods that rely on MAPI too?

Then we would need to change their types like that:

```js
export type FetchSpaceInfoError =
	| 'could-not-fetch-space-info'
	| 'region-not-identified'; <- no need to add this type here.
```

Would it be better? wdyt?

Reply:
@demetriusfeijoo 

Actually it looks like this, so it's a bit harder to add an extra type on the function declaration level like you suggested.

```
export type FetchSpaceInfo = (
	params: FetchSpaceInfoParams,
) => Promise<FetchSpaceInfoResponse>;

export type FetchSpaceInfoResponse =
	| { ok: true; result: SpaceInfo }
	| { ok: false; error: FetchSpaceInfoError };

export type FetchSpaceInfoError =
	| 'could-not-fetch-space-info'
	| 'region-not-identified';
```